services/table-foreign-keys-analyzer.js

```diff
@@ -8,50 +8,292 @@ function TableForeignKeysAnalyzer(databaseConnection, schema) {
-          JOIN information_schema.constraint_column_usage AS ccu
-            ON ccu.constraint_name = tc.constraint_name
-          WHERE constraint_type = 'FOREIGN KEY'
-            AND tc.table_name=:table;`;
+        SELECT
+          "tableConstraints".constraint_name AS "constraintName",
+          "tableConstraints".table_name AS "tableName",
+          "tableConstraints".constraint_type AS "columnType",
+          "keyColumnUsage".column_name AS "columnName",
+          "constraintColumnUsage".table_name AS "foreignTableName",
+          "constraintColumnUsage".column_name AS "foreignColumnName",
+          json_agg("uidx"."uniqueIndexes") filter (where "uidx"."uniqueIndexes" is not null) AS "uniqueIndexes"
+        FROM information_schema.table_constraints AS "tableConstraints"
+        JOIN information_schema.key_column_usage AS "keyColumnUsage"
+          ON "tableConstraints".constraint_name = "keyColumnUsage".constraint_name
+        JOIN information_schema.constraint_column_usage AS "constraintColumnUsage"
+          ON "constraintColumnUsage".constraint_name = "tableConstraints".constraint_name
+        FULL OUTER JOIN (
+          -- Get the index name, table name and list of columns of the unique indexes of a table
+          SELECT pg_index.indexrelid::regclass AS "indexName",
+            "pgClass1".relname AS "tableName",
+            json_agg(DISTINCT pg_attribute.attname) AS "uniqueIndexes"
+          FROM
+            pg_class AS "pgClass1",
+            pg_class AS "pgClass2",
+            pg_index,
+            pg_attribute
+          WHERE
+            "pgClass1".relname = :table
+            AND "pgClass1".oid = pg_index.indrelid
+            AND "pgClass2".oid = pg_index.indexrelid
+            AND pg_attribute.attrelid = "pgClass1".oid
+            AND pg_attribute.attnum = ANY(pg_index.indkey)
+            AND not pg_index.indisprimary
+            AND pg_index.indisunique
+            AND "pgClass1".relkind = 'r'
+            AND not "pgClass1".relname like 'pg%'
+            GROUP BY "tableName", "indexName"
+        ) AS "uidx"
+        ON "uidx"."tableName" = "tableConstraints".table_name
+        WHERE "uidx"."tableName" = :table OR "tableConstraints".table_name = :table
+        GROUP BY "constraintName", "tableConstraints".table_name, "columnType", "columnName", "foreignTableName", "foreignColumnName"`;
         break;
       case 'mysql':
         query = `
-          SELECT
-            TABLE_NAME AS table_name,
-            COLUMN_NAME AS column_name,CONSTRAINT_NAME AS constraint_name,
-            REFERENCED_TABLE_NAME AS foreign_table_name,
-            REFERENCED_COLUMN_NAME AS foreign_column_name
-          FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE
-          WHERE TABLE_SCHEMA = :databaseName
-            AND TABLE_NAME = :table;`;
-        replacements.databaseName = queryInterface.sequelize.config.database;
+        SELECT constraintName,
+               tableName,
```

Comment:
Not a fan of indentation depending of "prefix content". I used to do such things in my young days but it is a bad practice for maintenance IMO.
```suggestion
          tableName,
```
is better

Reply:
Same comment for all `SELECT` columns.